Remove deprecated tokenizers ConveRTTokenizer and LanguageModelTokenizer.

kedz opened the pull request:
**Proposed changes**:
Remove deprecated `LanguageModelTokenizer` and `ConveRTTokenizer` classes.
There were two tests in `tests/nlu/featurizers/test_lm_featurizer.py` that tested these tokenizers and I have removed them as well.
Docs entries were removed as well. 

Also removing `HFTransformersNLP` class, tests, and docs entries.

**Status (please check what you already did)**:
- [X] removed tokenizer class definitions.
- [X] updated tests 
- [X] updated the documentation
- [X] updated the changelog (please check [changelog](https://github.com/RasaHQ/rasa/tree/main/changelog) for instructions)
- [X] reformat files using `black` (please check [Readme](https://github.com/RasaHQ/rasa#code-style) for instructions)

wochinge commented on tests/nlu/test_train.py:
```diff
@@ -129,7 +129,6 @@ def test_all_components_are_in_at_least_one_test_pipeline():
 
     for cls in registry.component_classes:
         if "convert" in cls.name.lower():
-            # TODO
```
this is still a TODO, isn't it?

kedz replied:
thanks for catching that! Accidentally removed it. Its back now.

wochinge reviewed (approved):
Can we please also add a changelog item?

wochinge commented on docs/docs/components.mdx:
````diff
@@ -421,109 +421,6 @@ now implements its behavior.
     ```
 
 
-  ### ConveRTTokenizer
````
Good, good catch!

wochinge reviewed (approved):
💯

kedz commented:
@wochinge @dakshvar22 I can't figure out from the error messages why the Docs CI tests are failing. I tried to build the docs locally and am not having success even compiling the docs with our makefile. Do either of you have an idea of whats up there?

kedz commented:
/modeltest

```yml
include:
 - dataset: ["all"]
   config: ["all"]

```

kedz commented:
/modeltest

include:
 - dataset: ["all-nlu"]
   config: ["all-nlu"]

wochinge commented:
There are some `Changelog` items which link to the deleted docs sections. You need to remove this links from `Changelog.mdx`

kedz commented:
Hmm `changelog.mdx` is generated by something else and the offending links are from changes from a year ago but I can't figure out where the text of them lives. It isn't in `rasa/changelog` as far as I can tell. Do you know where it is?

kedz commented:
/modeltest

```yaml
include:
 - dataset: ["all-nlu"]
   config: ["all-nlu"]
```

kedz commented:
/modeltest

```yaml
dataset_branch: remove-deprecated-modules
include:
 - dataset: ["all-nlu"]
   config: ["all-nlu"]
```

kedz commented:
/modeltest

```yaml
dataset_branch: remove-deprecated-modules
include:
 - dataset: ["all-nlu"]
   config: ["all-nlu"]
```

kedz commented:
Looks like removing `LanguageModelTokenizer` breaks our regression tests 🙈. I created a [branch](https://github.com/RasaHQ/training-data/pull/16) of training-data that updates the configs that used deprecated components. @dakshvar22 I asked for a PR review from you on that branch since we will have to merge that before we can merge this to main. I'm running the NLU regression tests now on that branch. If they finish without changes in performance I believe we should be good to merge both this branch and the training-data branch.

dakshvar22 commented on changelog/8881.removal.md:
```diff
@@ -0,0 +1 @@
+Follow through on deprecation warnings and remove code, tests, and docs for `ConveRTTokenizer` and `LanguageModelTokenizer`.
```
I think we'll need to add an entry to the migration docs as well?

kedz replied:
`ConveRTTokenizer`, `LanguageModelToknizer`, and `HFTransformerNLP` modules have a migration entry [here](https://rasa.com/docs/rasa/migration-guide#deprecations-2) along with what modules replace their functionality. Do you think we need to add this somewhere else?

dakshvar22 replied:
Yeah, the migration guide will need a new section for all removals in Rasa 3.0 (basically a sub-section under a section like [this](https://rasa.com/docs/rasa/migration-guide#rasa-110-to-rasa-20) ).. `2.8 to 3.0`. Removal of these components should be mentioned there.
You might want to check with enable on how what's the format of the subsection they are willing to follow for the migration guide.

kedz replied:
I don't think we can write the 2.8 -> 3.0 migration guide yet as we haven't finalized what 3.0 configs look like. Perhaps this should be a separate issue that I should make? We should probably also involve CSE.

dakshvar22 replied:
> I don't think we can write the 2.8 -> 3.0 migration guide yet as we haven't finalized what 3.0 configs look like.

How does that matter? 😅  We can anyways start a section to add all the migration bits right? Is there anything specific from the config that you'll need to add the migration entry for this?

> We should probably also involve CSE.

You can ask them if they have already thought of a format, but mostly they might have not because we are getting started with the revamp. Leaving it out for a separate issue seems risky to me.

kedz replied:
> How does that matter? 😅 We can anyways start a section to add all the migration bits right? Is there anything specific from the config that you'll need to add the migration entry for this? 

Just looking at the existing migration docs now they have before and after config examples and so I'm not sure what to put there now for the after sections since we haven't yet decided what after looks like.

My larger concern is that we haven't done this for the policy removals either but that PR already got merged so I think its just as risky to create a migration guide as part of this issue and risk leaving that stuff out. 

Would consider this PR ok to signoff on if I made an issue aggregating what needs to go into migration docs and put it in our inbox/point cse to it to start discussion?

dakshvar22 replied:
> Would consider this PR ok to sign off on if I made an issue aggregating what needs to go into migration docs and put it in our inbox/point cse to it to start discussion?

👍

dakshvar22 commented on tests/nlu/featurizers/test_lm_featurizer.py:
```diff
@@ -655,52 +650,6 @@ def test_attention_mask(
-    with caplog.at_level(logging.DEBUG):
-        lm_featurizer.process(message)
-
-    assert "deprecated component HFTransformersNLP" in caplog.text
```
We haven't removed `HFTransformersNLP` as part of this PR, so this test should stay?

kedz replied:
Removed `HFTransformersNLP` module so it should be ok to remove this now.

dakshvar22 commented on tests/nlu/featurizers/test_lm_featurizer.py:
```diff
@@ -655,52 +650,6 @@ def test_attention_mask(
-    assert "deprecated component HFTransformersNLP" in caplog.text
-
-
-def test_preserve_sentence_and_sequence_features_old_config():
```
Same for this test..

kedz replied:
Removed `HFTransformersNLP` module so it should be ok to remove this now.

kedz commented:
/modeltest

```yaml
include:
 - dataset: ["all-nlu"]
   config: ["all-nlu"]
```

dakshvar22 commented on docs/docs/components.mdx:
````diff
@@ -130,97 +130,6 @@ word vectors in your pipeline.
-  [HuggingFace documentation](https://huggingface.co/transformers/pretrained_models.html).
-  If left empty, it uses the default model architecture that original Transformers library loads (see table below).
-
-  ```
-  +----------------+--------------+-------------------------+
-  | Language Model | Parameter    | Default value for       |
-  |                | "model_name" | "model_weights"         |
-  +----------------+--------------+-------------------------+
-  | BERT           | bert         | rasa/LaBSE              |
-  +----------------+--------------+-------------------------+
-  | GPT            | gpt          | openai-gpt              |
-  +----------------+--------------+-------------------------+
-  | GPT-2          | gpt2         | gpt2                    |
-  +----------------+--------------+-------------------------+
-  | XLNet          | xlnet        | xlnet-base-cased        |
-  +----------------+--------------+-------------------------+
-  | DistilBERT     | distilbert   | distilbert-base-uncased |
-  +----------------+--------------+-------------------------+
-  | RoBERTa        | roberta      | roberta-base            |
-  +----------------+--------------+-------------------------+
-  ```
````
Do we have this info already in docs for `LanguageModelFeaturizer`? If not, we should add it. Can't see it in the truncated diff here.

kedz replied:
Yes this table already exists in the docs for `LanguageModelFeaturizer`.

dakshvar22 commented on docs/docs/tuning-your-model.mdx:
```diff
@@ -219,8 +219,6 @@ We support a few components that provide pre-trained word embeddings:
 
 3. [ConveRTFeaturizer](./components.mdx#convertfeaturizer)
 
-4. [LanguageModelFeaturizer](./components.mdx#languagemodelfeaturizer)
-
```
Why should this be removed?

dakshvar22 reviewed (changes requested):
Small comment on the docs but also need to remove references to `HFTransformersNLP` and `LanguageModelFeaturizer` in `tests/train.py`.

dakshvar22 commented:
Also, there is a reference to `ConveRTTokenizer` in `tests/nlu/conftest.py` that should be removed.

dakshvar22 commented on changelog/8881.removal.md:
```diff
@@ -0,0 +1,4 @@
+Follow through on deprecation warnings and remove code, tests, and docs for `ConveRTTokenizer` and `LanguageModelTokenizer`.
+
+
+Follow through on deprecation warnings and remove code, tests, and docs for `HFTransformersNLP`.
```
Optional: You could just combine it into 1 😅 
```suggestion
Follow through on deprecation warnings and remove code, tests, and docs for `ConveRTTokenizer`, `LanguageModelTokenizer` and `HFTransformersNLP`.
```

dakshvar22 commented:
@kedz There's still a reference to `HFTransformersNLP` in `tests/train.py`

kedz commented:
@dakshvar22 I removed the last reference to `HFTransformersNLP` in `tests/train.py` (thought I had removed it before 😅) and grepped the code base for that and the tokenizers and I think everything has been removed. 

The issue for the migration guide is [here](https://github.com/RasaHQ/rasa/issues/9186). On your approval I'll merge this PR.

dakshvar22 reviewed (approved):
💯
